lib/dirac_coarse.cpp

```diff
@@ -33,10 +33,10 @@ namespace quda {
     initializeCoarse();
   }
 
-  DiracCoarse::DiracCoarse(const DiracParam &param, cpuGaugeField *Y_h, cpuGaugeField *X_h, cpuGaugeField *Xinv_h,
-                           cpuGaugeField *Yhat_h, // cpu link fields
-                           cudaGaugeField *Y_d, cudaGaugeField *X_d, cudaGaugeField *Xinv_d,
-                           cudaGaugeField *Yhat_d) // gpu link field
+  DiracCoarse::DiracCoarse(const DiracParam &param, GaugeField *Y_h, GaugeField *X_h, GaugeField *Xinv_h,
+                           GaugeField *Yhat_h, // cpu link fields
+                           GaugeField *Y_d, GaugeField *X_d, GaugeField *Xinv_d,
+                           GaugeField *Yhat_d) // gpu link field
```

Comment:
With the MMA PR merged, these pointers are now `shared_ptr` so there are some merge needed here ...

Reply:
71a5335284050cc416a4937ca57e72fd990c680b